db/migrate/20180617074552_add_communication_opt_in_to_users.rb

```diff
@@ -1,10 +1,16 @@
 class AddCommunicationOptInToUsers < ActiveRecord::Migration[5.1]
   def change
     add_column :users, :opted_in_newsletter, :boolean, default: false
+    add_column :users, :opted_in_newsletter_at, :datetime
```

Comment:
I think what @carpodaster meant was: have a `datetime` column **instead** of the `boolean` one, not both.

The previous boolean value for `false` would then just be `NULL` with the `datetime`column.

Same as e.g. you'd have `user.confirmed?`  just check for a value on `user.confirmed_at`, instead of both being separate columns on the DB (which you'd have to keep in sync which is a way harder problem 😉 )